Item/Item.module.less

```diff
@@ -105,43 +107,95 @@
 		&.inline {
 			// overriding ui/Item's inline-block
 			display: inline-flex;
+			min-width: @agate-item-inline-min-width;
 			max-width: @agate-item-inline-max-width;
 
 			.itemContent {
 				max-width: @agate-item-inline-content-max-width;
 			}
 
+			&:not(.centered) {
+				.itemContent {
+					min-width: @agate-item-inline-notcentered-itemcontent-min-width;
+				}
+			}
+
 			.itemContent,
 			.content {
 				flex-shrink: 1;
-				overflow: auto;
-			}
-
-			&.hasLabel {
-				.slotAfter {
-					.padding-start-end(@agate-item-slotafter-padding-left, 0);
-				}
+				overflow: hidden;
 			}
 		}
 
 		&.centered {
+			// override Marquee's alignment.
+			.marquee {
+				text-align: center;
+			}
+
 			.itemContent {
-				max-width: fit-content;
+				max-width: 100%;
+
+				&.labelAfter,
+				&.labelBefore {
+					.content {
+						min-width: @agate-item-content-width;
+						max-width: @agate-item-content-width;
+					}
+					.label {
+						max-width: @agate-item-label-width;
+					}
+				}
+
+				&.labelAfter {
+					.content {
+						.marquee {
+							text-align: right;
+						}
+					}
+					.label {
+						.marquee {
+							text-align: left;
+						}
+					}
+				}
+
+				&.labelBefore {
+					.content {
+						.marquee {
+							text-align: left;
+						}
+					}
+					.label {
+						.marquee {
+							text-align: right;
+						}
+					}
+				}
 			}
 
 			&.inline {
-				max-width: @agate-item-inline-max-width;
+				max-width: @agate-item-centered-inline-max-width;
+
+				.itemContent {
+					width: 100%;
+				}
 			}
 		}
 
-		&.large { /* Needed to prevent global class being added in the DOM */ }
+		&.large {
+			/* Available for export */
+		}
 
 		// TODO: needs to be modified when new requirements for small item is available
 		&.small {
+			height: @agate-item-small-height;
 			font-size: @agate-item-small-font-size;
 			line-height: @agate-item-small-line-height;
 
 			&.hasLabel {
+				height: @agate-item-small-haslabel-height;
```

Comment:
Is this needed?

Reply:
If we remove this line, there would be a layout bug on Gallium skin when small Item has a label, as seen below:
![small-height-gallium](https://user-images.githubusercontent.com/63335823/113128533-90c05800-9222-11eb-824a-219e7efe8266.PNG)

We also need a variable for Silicon because the height of small item with label will grow over 99px if this line is removed

Reply:
🆗